hawkbit-artifact-repository-filesystem/src/main/java/org/eclipse/hawkbit/artifact/repository/ArtifactFilesystem.java

```diff
@@ -35,6 +38,10 @@
     // cannot be closed in this method
     @SuppressWarnings("squid:S2095")
     public InputStream getFileInputStream() {
+        if (file == null) {
```

Comment:
I think it would be better for the `getFileInputStream()` to throw an exception in case the inputstream cannot be opened instead of returning `null`, because a caller would not check for return value `null` when opening an inpustream.